### PR TITLE
List team news

### PR DESCRIPTION
**Pull request by gary-van-woerkens**
- List latest news on landing page
- List team news in dedicated page
- Improve publish form

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -2,81 +2,35 @@ import React from "react";
...
 }
 
 Home.getInitialProps = async ({ req }) => {
-  // fetch data for given month
-  const now = new Date();
-  const { year = now.getFullYear(), month = now.getMonth() + 1 } =
-    (req && req.query) || {};
-
-  let url = `/api/fetchPosts?year=${year}&month=${month}`;
+  let url = `/api/fetchLatestPosts`;
```
Comment (revolunet) [inline pages/index.js]: i guess we could directly call the method from the api route here so we save one internal request

File: pages/teams/[team].js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  const payload = await fetch(url);
+  const data = await payload.json();
+  data.sort(
+    (a, b) =>
+      new Date(b.content.date).getTime() - new Date(a.content.date).getTime()
+  );
+
+  return { data };
```
Comment (revolunet) [inline pages/teams/[team].js]: you can pass `team: query.team` here so you dont need to require router in the next component

File: src/components/FormPublish.js [hunk 3]
```diff
@@ -81,9 +94,8 @@ const FormPublish = ({ onSubmit }) => {
         return errors;
       }}
       onSubmit={(values, { setSubmitting }) => {
-        setTimeout(() => {
-          alert(JSON.stringify(values, null, 2));
-          onSubmit(values);
+        setTimeout(async () => {
```
Comment (revolunet) [inline src/components/FormPublish.js]: shoudn't we have `setSubmitting(true)` as first action ?

Comment (revolunet) [reply src/components/FormPublish.js]: also why 400 ?

File: src/components/FormPublish.js [hunk 4]
```diff
@@ -212,15 +225,33 @@ const FormPublish = ({ onSubmit }) => {
...
-                size="lg"
-                block
-                color="primary"
-                onClick={handleSubmit}
-              >
-                Publier
-              </Button>
+              {!isSubmitting ? (
+                <Button
+                  disabled={isSubmitting}
```
Comment (revolunet) [inline src/components/FormPublish.js]: useless attribute

File: src/lib/teams.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    .replace(/[\s/;:,!?#]/g, "-")
+    .toLowerCase();
+
+teams = teams.reduce((teams, team) => {
+  teams.push({
+    name: team,
+    id: sanitize(team)
+  });
+  return teams;
+}, []);
```
Comment (revolunet) [inline src/lib/teams.js]: [map](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/map) is your friend 

also for sanitize we could use [some slugify package](https://github.com/sindresorhus/slugify)

File: pages/index.js (hunk 1 above)
Comment (gary-van-woerkens) [reply pages/index.js]: Alright. I must refactor the 2 calls fetching the published posts anyway :)

File: pages/teams/[team].js (hunk 2 above)
Comment (gary-van-woerkens) [reply pages/teams/[team].js]: Yep, true... since I get it in the `getInitialProps`.

File: src/components/FormPublish.js (hunk 3 above)
Comment (gary-van-woerkens) [reply src/components/FormPublish.js]: Obviously the `setSubmitting(true)` is done under the curtains. I've tested submitting the form 2 times in a row and I can see that the `isSubmitting` flag works properly.
Still, it's probably a good idea to set it back explicitly each time.

Comment (gary-van-woerkens) [reply src/components/FormPublish.js]: About the `400`, no idea why it is there... gonna change it.

File: src/components/FormPublish.js (hunk 4 above)
Comment (gary-van-woerkens) [reply src/components/FormPublish.js]: True!

File: src/lib/teams.js (hunk 5 above)
Comment (gary-van-woerkens) [reply src/lib/teams.js]: I should replace this `reduce` by a `map`... okay, gonna try it out.
I like our light `sanitize` :)

File: pages/index.js (hunk 1 above)
Comment (gary-van-woerkens) [reply pages/index.js]: I've open a specific issue for this one: [issue 13](https://github.com/SocialGouv/carnets/issues/13)

File: src/components/FormPublish.js (hunk 3 above)
Comment (gary-van-woerkens) [reply src/components/FormPublish.js]: I've open a specific issue for this one: [issue 12](https://github.com/SocialGouv/carnets/issues/12)

**Pull request merged**